### ISSUE TITLE
Test `jax.lax.stop_gradient()` for PlaneType enum class

### ISSUE THREAD

**Issue by LouisDesdoigts** [LouisDesdoigts/dLux#107]
So introducing the PlaneType enum object has mandated passing in the `eqx.is_inexact_array` filter function to all calls of `eqx.filter_jit` which is not ideal. Patrick Kidger suggested that we could use the ``jax.lax.stop_gradient()` function to prevent gradients being propagated through this class. He provided a skeleton example below:

![IMG_8048](https://user-images.githubusercontent.com/42196907/186809674-3be393c7-e26a-462c-b1f4-7dc50d2ab07c.jpg)
